DB/src/main/java/io/deephaven/db/tables/utils/ParquetTools.java

```diff
@@ -0,0 +1,409 @@
+        } catch (ClassNotFoundException e) {
+            throw new UncheckedDeephavenException(
+                    "Column " + colName + " with " + desc + "=" + className + " that can't be found in classloader.");
+        }
+    }
+
+    private static ParquetReaderUtil.ColumnDefinitionConsumer makeSchemaReaderConsumer(final ArrayList<ColumnDefinition> cols) {
+        return (final String name, final Class<?> typeFromParquet, final String dbSpecialType, final boolean isLegacyType, final boolean isArray,
+                final boolean isGrouping, final String codecName, final String codecArgs, final String codecType, final String codecComponentType) -> {
+            Class<?> baseType;
+            if (typeFromParquet != null && typeFromParquet.equals(boolean.class)) {
+                baseType = Boolean.class;
+            } else {
+                baseType = typeFromParquet;
+            }
+            final ColumnDefinition<?> colDef;
+            if (codecName != null) {
+                Class<?> dataType = baseType;
+                Class<?> componentType = null;
+                if (codecType != null && !codecType.isEmpty()) {
+                    if (codecComponentType != null && !codecComponentType.isEmpty()) {
+                        componentType = loadClass(name, "codecComponentType", codecComponentType);
+                    }
+                    dataType = loadClass(name, "codecType", codecType);
+                }
+                final ObjectCodec<?> codec = CodecCache.DEFAULT.getCodec(codecName, codecArgs);
+                final int width = codec.expectedObjectWidth();
+                if (width != ObjectDecoder.VARIABLE_WIDTH_SENTINEL) {
+                    colDef = ColumnDefinition.ofFixedWidthCodec(name, dataType, componentType, codecName, codecArgs, width);
+                } else {
+                    colDef = ColumnDefinition.ofVariableWidthCodec(name, dataType, componentType, codecName, codecArgs);
+                }
+            } else if (dbSpecialType != null) {
+                if (dbSpecialType.equals(ParquetTableWriter.STRING_SET_SPECIAL_TYPE)) {
+                    colDef = ColumnDefinition.fromGenericType(name, StringSet.class, null);
+                } else if (dbSpecialType.equals(ParquetTableWriter.DBARRAY_SPECIAL_TYPE)) {
+                    final Class<?> dbArrayType = dbArrayType(baseType);
+                    if (dbArrayType != null) {
+                        colDef = ColumnDefinition.fromGenericType(name, dbArrayType, baseType);
+                    } else {
+                        colDef = ColumnDefinition.fromGenericType(name, DbArray.class, baseType);
+                    }
+                } else {
+                    throw new UncheckedDeephavenException("Unhandled dbSpecialType=" + dbSpecialType);
+                }
+            } else {
+                if (!StringSet.class.isAssignableFrom(baseType) && isArray) {
+                    if (baseType.equals(byte.class) && isLegacyType) {
```

Comment:
`isLegacyType` -> `isUnspecifiedBinaryType`?

Reply:
That's not what it means, if you look at the code in ParquetReadUtil that produces it.  It literally means "this came from old parquet without logic type annotations"